### PR TITLE
sql: add doc for `flashback database`

### PR DESCRIPTION
**Pull request by Defined2014**
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):
feature link https://github.com/pingcap/tidb/pull/30217
ref https://docs.pingcap.com/zh/tidb/stable/sql-statement-flashback-table

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sql-statements/sql-statement-flashback-database.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+---
+title: FLASHBACK DATABASE
+summary: TiDB 数据库中 FLASHBACK DATABASE 的使用概况。
+---
+
+# FLASHBACK DATABASE
```
Comment (ran-huang) [inline sql-statements/sql-statement-flashback-database.md:6]: 这是实验特性还是 GA 呢？

File: sql-statements/sql-statement-flashback-database.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+在 TiDB 6.4 中，引入了 `FLASHBACK DATABASE` 语法，其功能是在 Garbage Collection (GC) life time 时间内，可以用 `FLASHBACK DATABASE` 语句来恢复被 `DROP` 删除的数据库以及数据。
```
Comment (ran-huang) [inline sql-statements/sql-statement-flashback-database.md:8]:
```suggestion
TiDB v6.4.0 引入了 `FLASHBACK DATABASE` 语法，其功能是在 Garbage Collection (GC) life time 时间内，可以用 `FLASHBACK DATABASE` 语句来恢复被 `DROP` 删除的数据库以及数据。
```

File: sql-statements/sql-statement-flashback-database.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+### 语法图
+
+```ebnf+diagram
+FlashbackDatabaseStmt ::=
+    'FLASHBACK' DatabaseSym DBName FlashbackToNewName
+
+FlashbackToNewName ::=
+    ( 'TO' Identifier )?
+```
```
Comment (ran-huang) [inline sql-statements/sql-statement-flashback-database.md:36]: 麻烦确认下语法图渲染出来是正确的～
image

File: sql-statements/sql-statement-flashback-database.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+可以使用系统变量 [`tidb_gc_life_time`](/system-variables.md#tidb_gc_life_time-从-v50-版本开始引入) 配置数据的历史版本的保留时间（默认值是 `10m0s`）。可以使用以下 SQL 语句查询当前的 `safePoint`，即 GC 已经清理到的时间点：
+
+{{< copyable "sql" >}}
+
+  ```sql
+  SELECT * FROM mysql.tidb WHERE variable_name = 'tikv_gc_safe_point';
+  ```
+
+只要被 `DROP` 数据库是在 `tikv_gc_safe_point` 时间之后，都能用 `FLASHBACK DATABASE` 语法来恢复。
```
Comment (ran-huang) [inline sql-statements/sql-statement-flashback-database.md:18]:
```suggestion
只要一个数据库是在 `tikv_gc_safe_point` 时间之后被 `DROP`，都能用 `FLASHBACK DATABASE` 语法来恢复。
```

File: sql-statements/sql-statement-flashback-database.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+FlashbackDatabaseStmt ::=
+    'FLASHBACK' DatabaseSym DBName FlashbackToNewName
+
+FlashbackToNewName ::=
+    ( 'TO' Identifier )?
+```
+
+## 注意事项
+
+如果删除了一个数据库并过了 GC lifetime，就不能再用 `FLASHBACK DATABASE` 语句来恢复被删除的数据了，否则会返回错误，错误类似于 `ERROR 1105 (HY000): Can't find dropped database 'test' in GC safe point 2022-11-06 16:10:10 +0800 CST`。
```
Comment (ran-huang) [inline sql-statements/sql-statement-flashback-database.md:40]:
```suggestion
如果数据库被删除的时间超过了 GC lifetime (`tikv_gc_safe_point`)，就无法使用 `FLASHBACK DATABASE` 语句来恢复被删除的数据了，否则会返回错误，错误类似于 `ERROR 1105 (HY000): Can't find dropped database 'test' in GC safe point 2022-11-06 16:10:10 +0800 CST`。
```

**Review by xhebox — APPROVED** [review]

File: sql-statements/sql-statement-flashback-database.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+不能用 `FLASHBACK DATABASE` 多次恢复同一个被删除的数据库，因为 `FLASHBACK DATABASE` 所恢复数据库的 schema ID 还是被删除数据库的 schema ID，而 TiDB 要求所有还存在的数据库 schema ID 必须全局唯一。
```
Comment (ran-huang) [inline sql-statements/sql-statement-flashback-database.md:42]: 这样理解正确吗？
```suggestion
不能用 `FLASHBACK DATABASE` 多次恢复同一个被删除的数据库，因为 `FLASHBACK DATABASE` 所恢复数据库的 schema ID 还是被删除数据库的 schema ID，多次恢复同一个被删除的数据库会导致重复的 schema ID。在 TiDB 中，所有还存在的数据库 schema ID 必须全局唯一。
```

File: sql-statements/sql-statement-flashback-database.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+在开启 TiDB Binlog 时使用 `FLASHBACK DATABASE` 需要注意以下情况：
+
+* 下游从集群也支持 `FLASHBACK DATABASE`
```
Comment (ran-huang) [inline sql-statements/sql-statement-flashback-database.md:46]:
```suggestion
* 下游从集群也支持 `FLASHBACK DATABASE`。
```

File: sql-statements/sql-statement-flashback-database.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+* 从集群的 GC lifetime 一定要长于主集群的 GC lifetime。上下游同步存在的延迟可能也会造成下游恢复数据失败。
```
Comment (ran-huang) [inline sql-statements/sql-statement-flashback-database.md:47]:
```suggestion
* 从集群的 GC lifetime 一定要长于主集群的 GC lifetime。否则上下游同步存在的延迟可能也会造成下游恢复数据失败。
```

File: sql-statements/sql-statement-flashback-database.md [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+如果 Binlog 同步出错，则需要在 Binlog 过滤掉该数据库，同时手动全量重新导入该数据库的数据。
```
Comment (ran-huang) [inline sql-statements/sql-statement-flashback-database.md:49]:
```suggestion
如果 TiDB Binlog 同步出错，则需要在 TiDB Binlog 中过滤掉该数据库，同时手动全量重新导入该数据库的数据。
```

File: sql-statements/sql-statement-flashback-database.md (hunk 3 above)
Comment (Defined2014) [reply sql-statements/sql-statement-flashback-database.md:36]: 没问题

File: sql-statements/sql-statement-flashback-database.md (hunk 1 above)
Comment (Defined2014) [reply sql-statements/sql-statement-flashback-database.md:6]: GA

**Review by ran-huang — APPROVED** [review]

File: sql-statements/sql-statement-flashback-database.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+## 注意事项
+
+- 如果数据库被删除的时间超过了 GC lifetime (`tikv_gc_safe_point`)，就无法使用 `FLASHBACK DATABASE` 语句来恢复被删除的数据了，否则会返回错误。错误类似于 `ERROR 1105 (HY000): Can't find dropped database 'test' in GC safe point 2022-11-06 16:10:10 +0800 CST`。
+
+- 不能用 `FLASHBACK DATABASE` 多次恢复同一个被删除的数据库，因为 `FLASHBACK DATABASE` 所恢复数据库的 schema ID 还是被删除数据库的 schema ID，多次恢复同一个被删除的数据库会导致重复的 schema ID。在 TiDB 中，所有还存在的数据库 schema ID 必须全局唯一。
+
+- 在开启 TiDB Binlog 时，使用 `FLASHBACK DATABASE` 需要注意以下情况：
+
+    * 下游从集群也支持 `FLASHBACK DATABASE`。
```
Comment (hfxsd) [inline sql-statements/sql-statement-flashback-database.md:42]:
```suggestion
    * 下游从集群也需要支持 `FLASHBACK DATABASE`。
```

File: sql-statements/sql-statement-flashback-database.md [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+    ```
+
+- 恢复被 `DROP` 删除的 `test` 数据库并重命名为 `test1`：
+
+    ```sql
+    DROP DATABASE test;
+    ```
+
+    ```sql
+    FLASHBACK DATABASE test to test1;
```
Comment (hfxsd) [inline sql-statements/sql-statement-flashback-database.md:65]: TO 是不是应该大写？
```suggestion
    FLASHBACK DATABASE test TO test1;
```

File: sql-statements/sql-statement-flashback-database.md [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+## 注意事项
+
+- 如果数据库被删除的时间超过了 GC lifetime (`tikv_gc_safe_point`)，就无法使用 `FLASHBACK DATABASE` 语句来恢复被删除的数据了，否则会返回错误。错误类似于 `ERROR 1105 (HY000): Can't find dropped database 'test' in GC safe point 2022-11-06 16:10:10 +0800 CST`。
+
+- 不能用 `FLASHBACK DATABASE` 多次恢复同一个被删除的数据库，因为 `FLASHBACK DATABASE` 所恢复数据库的 schema ID 还是被删除数据库的 schema ID，多次恢复同一个被删除的数据库会导致重复的 schema ID。在 TiDB 中，所有还存在的数据库 schema ID 必须全局唯一。
+
+- 在开启 TiDB Binlog 时，使用 `FLASHBACK DATABASE` 需要注意以下情况：
+
+    * 下游从集群也支持 `FLASHBACK DATABASE`。
+    * 从集群的 GC lifetime 一定要长于主集群的 GC lifetime。否则上下游同步存在的延迟可能也会造成下游恢复数据失败。
```
Comment (hfxsd) [inline sql-statements/sql-statement-flashback-database.md:43]:
```suggestion
    * 从集群的 GC lifetime 一定要长于主集群的 GC life time。否则上下游同步存在的延迟可能也会造成下游恢复数据失败。
```

File: sql-statements/sql-statement-flashback-database.md [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+FlashbackDatabaseStmt ::=
+    'FLASHBACK' DatabaseSym DBName FlashbackToNewName
+
+FlashbackToNewName ::=
+    ( 'TO' Identifier )?
+```
+
+## 注意事项
+
+- 如果数据库被删除的时间超过了 GC lifetime (`tikv_gc_safe_point`)，就无法使用 `FLASHBACK DATABASE` 语句来恢复被删除的数据了，否则会返回错误。错误类似于 `ERROR 1105 (HY000): Can't find dropped database 'test' in GC safe point 2022-11-06 16:10:10 +0800 CST`。
```
Comment (hfxsd) [inline sql-statements/sql-statement-flashback-database.md:36]: lifetime 和 life time 混用。都统一成 life time 吧
```suggestion
- 如果数据库被删除的时间超过了 GC life time (`tikv_gc_safe_point`)，就无法使用 `FLASHBACK DATABASE` 语句来恢复被删除的数据了，否则会返回错误。错误类似于 `ERROR 1105 (HY000): Can't find dropped database 'test' in GC safe point 2022-11-06 16:10:10 +0800 CST`。
```

File: sql-statements/sql-statement-flashback-database.md [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+
+- 不能用 `FLASHBACK DATABASE` 多次恢复同一个被删除的数据库，因为 `FLASHBACK DATABASE` 所恢复数据库的 schema ID 还是被删除数据库的 schema ID，多次恢复同一个被删除的数据库会导致重复的 schema ID。在 TiDB 中，所有还存在的数据库 schema ID 必须全局唯一。
```
Comment (hfxsd) [inline sql-statements/sql-statement-flashback-database.md:38]: “还存在的数据库” 这里的 “还存在的” 是什么意思？考虑换个更好理解的说法吧

Comment (Defined2014) [inline sql-statements/sql-statement-flashback-database.md:38]:
```suggestion
- 不能用 `FLASHBACK DATABASE` 多次恢复同一个被删除的数据库，因为 `FLASHBACK DATABASE` 所恢复数据库的 schema ID 和原被删除数据库的 schema ID 一致，多次恢复同一数据库会导致重复的 schema ID。在 TiDB 中，所有数据库的 schema ID 必须全局唯一。
```

File: sql-statements/sql-statement-flashback-database.md [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+## 注意事项
+
+- 如果数据库被删除的时间超过了 GC life time (`tikv_gc_safe_point`)，就无法使用 `FLASHBACK DATABASE` 语句来恢复被删除的数据了，否则会返回错误。错误类似于 `ERROR 1105 (HY000): Can't find dropped database 'test' in GC safe point 2022-11-06 16:10:10 +0800 CST`。
+
+- 不能用 `FLASHBACK DATABASE` 多次恢复同一个被删除的数据库，因为 `FLASHBACK DATABASE` 所恢复数据库的 schema ID 和原被删除数据库的 schema ID 一致，多次恢复同一数据库会导致重复的 schema ID。在 TiDB 中，所有数据库的 schema ID 必须全局唯一。
+
+- 在开启 TiDB Binlog 时，使用 `FLASHBACK DATABASE` 需要注意以下情况：
+
+    * 下游从集群也需要支持 `FLASHBACK DATABASE`。
+    * 从集群的 GC lifetime 一定要长于主集群的 GC life time。否则上下游同步存在的延迟可能也会造成下游恢复数据失败。
```
Comment (Defined2014) [inline sql-statements/sql-statement-flashback-database.md:43]:
```suggestion
    * 从集群的 GC life time 一定要长于主集群的 GC life time。否则上下游同步存在的延迟可能也会造成下游恢复数据失败。
```

**Review by hfxsd — APPROVED** [review]

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**